.github/workflows/test.yml

```
@@ -65,24 +69,42 @@ jobs:
 
   django:
     needs: functional
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         # https://docs.djangoproject.com/en/3.2/faq/install/#what-python-version-can-i-use-with-django
         # this is a bit excessive, probably should run only before
         #   releasing new version
-        python-version: [3.6, 3.7, 3.8, 3.9]
-        django-version: [2.2, 3.0, 3.1, 3.2]
+        os: [ubuntu-20.04, ubuntu-latest]
+        python-version: ['3.6', '3.7', '3.8', '3.9']
+        django-version: ['2.2', '3.0', '3.1', '3.2']
```

this is not good.
I'd rather make explicit includes for 3.6. It will be much easier to maintain in the future (and remove as well).
```suggestion
        os: [ubuntu-latest]
        python-version: ['3.7', '3.8', '3.9']
        django-version: ['2.2', '3.0', '3.1', '3.2']
```

---

as an option, move it to separate step "functional-legacy-py" so we could simplify matrix here and there?

---

I decided to keep `include` section with py36 in `functional` matrix because it's rather straightforward and simply extend matrix with one extra environment set. And I fully agree with you about `django` matrix, it is not so obvious with `exclude+include` parts, so I have split it into 2: `django` for supported python versions and `django-legacy-py` with py36 only, but this matrix definitely will grow in the nearest future - `py37` is supported till [June 2023 ](https://endoflife.date/python)